artemis-core-client/src/main/java/org/apache/activemq/artemis/spi/core/protocol/RemotingConnection.java

```diff
@@ -245,6 +245,13 @@
     */
    String getTransportLocalAddress();
 
+   /**
+    * Called when all connection related operations are completed.
+    */
+   default void connected() throws Exception {
```

Comment:
As this introduces a notification about connections created, should ensure that this is always called, and not dependent on a protocol logic.